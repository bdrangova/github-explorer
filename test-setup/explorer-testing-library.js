import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import rootReducer from 'src/store';
import { messages } from 'src/locales';

import { FirebaseAuth } from 'src/providers/Auth';

const renderWithRouter = (
  ui,
  { route = '/', renderOptions = {} } = {},
) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const utils = render(renderRouter(ui, history), renderOptions);

  return {
    ...utils,
    history,
  };
};

const renderRouter = (ui, history) => {
  return <Router history={history}>{ui}</Router>;
};

const renderWithRedux = (
  ui,
  { initialState = {}, renderOptions = {} } = {},
) => {
  const utils = render(
    renderRedux(ui, initialState),
    ...renderOptions,
  );
  return {
    ...utils,
  };
};

const renderRedux = (ui, initialState = {}) => {
  const store = createStore(rootReducer, initialState);
  return <Provider store={store}>{ui}</Provider>;
};

const renderWithAuth = (
  ui,
  { value = {}, renderOptions = {} } = {},
) => {
  const utils = render(renderAuth(ui, value));
  return { ...utils };
};

const renderAuth = (ui, value = {}) => {
  return (
    <FirebaseAuth.Provider value={value}>{ui}</FirebaseAuth.Provider>
  );
};

const renderWithIntl = (ui, { renderOptions = {} } = {}) => {
  const utils = render(renderIntl(ui, renderOptions), renderOptions);
  return { ...utils };
};

const renderIntl = (ui) => {
  return (
    <IntlProvider locale="en" messages={messages['en']}>
      {ui}
    </IntlProvider>
  );
};

const renderFullstack = (
  ui,
  initialState = {},
  route = '/',
  value = {},
) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  return renderRedux(
    renderIntl(renderRouter(renderAuth(ui, value), history)),
    initialState,
  );
};

const renderWithFullstack = (
  ui,
  initialState = {},
  route = '/',
  authValue = {},
  renderOptions = {},
) => {
  return render(
    renderFullstack(ui, initialState, route, authValue),
    renderOptions,
  );
};

const renderWithIntlAndRouter = (
  ui,
  route = '/',
  renderOptions = {},
) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  return renderWithIntl(renderRouter(ui, history), renderOptions);
};

export {
  renderRouter,
  renderRedux,
  renderAuth,
  renderIntl,
  renderWithRouter,
  renderWithRedux,
  renderWithAuth,
  renderWithIntl,
  renderWithIntlAndRouter,
  renderWithFullstack,
};
