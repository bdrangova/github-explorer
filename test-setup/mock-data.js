const RepositoryMock = {
  id: 1234,
  name: 'react',
  full_name: 'facebook/react',
  description: 'Library for building UIs',
  stargazers_count: 150000,
  language: 'JavaScript',
  updated_at: '',
  owner: {
    login: 'facebook',
  },
  private: false,
};

const RepositoryArrayMock = [
  {
    id: 1234,
    name: 'react',
    full_name: 'facebook/react',
    description: 'Library for building UIs',
    stargazers_count: 150000,
    language: 'JavaScript',
    updated_at: '',
    owner: {
      login: 'facebook',
    },
    private: false,
  },
  {
    id: 46456,
    name: 'vue',
    full_name: 'vuejs/vue',
    description: 'Progressive framework',
    stargazers_count: 16000,
    language: 'JavaScript',
    updated_at: '',
    owner: {
      login: 'vuejs',
    },
    private: false,
  },
];

const MockedGithubSearchResponse = {
  total_count: 2,
  incomplete_results: false,
  items: [
    {
      id: 11730342,
      name: 'vue',
      full_name: 'vuejs/vue',
      private: false,
      owner: {
        login: 'vuejs',
        id: 6128107,
        avatar_url:
          'https://avatars1.githubusercontent.com/u/6128107?v=4',
      },
      html_url: 'https://github.com/vuejs/vue',
      description:
        'üññ Vue.js is a progressive, incrementally-adoptable JavaScript framework for building UI on the web.',
      updated_at: '2020-06-02T08:29:16Z',

      stargazers_count: 165128,
      watchers_count: 165128,
      language: 'JavaScript',
      forks_count: 25032,
      open_issues_count: 509,
      forks: 25032,
    },
    {
      id: 77189043,
      name: 'vue2-elm',
      full_name: 'bailicangdu/vue2-elm',
      private: false,
      owner: {
        login: 'bailicangdu',
        id: 20297227,
        avatar_url:
          'https://avatars2.githubusercontent.com/u/20297227?v=4',
      },
      html_url: 'https://github.com/bailicangdu/vue2-elm',
      description:
        'Âü∫‰∫é vue2 + vuex ÊûÑÂª∫‰∏Ä‰∏™ÂÖ∑Êúâ 45 ‰∏™È°µÈù¢ÁöÑÂ§ßÂûãÂçïÈ°µÈù¢Â∫îÁî®',
      updated_at: '2020-06-02T08:13:35Z',
      size: 29812,
      stargazers_count: 33331,
      watchers_count: 33331,
      language: 'Vue',
      forks_count: 11052,
      open_issues_count: 72,
    },
  ],
};

export {
  RepositoryMock,
  RepositoryArrayMock,
  MockedGithubSearchResponse,
};
