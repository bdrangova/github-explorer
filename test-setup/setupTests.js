import '@testing-library/jest-dom/extend-expect';
require('jest-fetch-mock').enableMocks();
require('../__mocks__/browserMocks');
