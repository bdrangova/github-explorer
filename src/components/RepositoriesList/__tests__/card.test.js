import React from 'react';
import { renderWithIntlAndRouter } from 'test-setup/explorer-testing-library';
import { RepositoryMock } from 'test-setup/mock-data';
import Card from '../card';

describe('Repository card item', () => {
  it('renders properties correctly', () => {
    const { getByTestId } = renderWithIntlAndRouter(
      <Card item={RepositoryMock} />,
    );

    const ownerName = getByTestId('owner-name');
    const repoName = getByTestId('repo-name');

    expect(ownerName).toHaveTextContent(RepositoryMock.owner.login);
    expect(repoName).toHaveTextContent(RepositoryMock.name);
  });

  it('repository card snapshot', () => {
    const { container } = renderWithIntlAndRouter(
      <Card item={RepositoryMock} />,
    );

    expect(container).toMatchSnapshot();
  });
});
