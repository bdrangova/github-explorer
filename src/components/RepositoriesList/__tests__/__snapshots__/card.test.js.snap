// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repository card item repository card snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root card"
      tabindex="0"
      type="button"
    >
      <a
        href="/repos/facebook/react"
        style="text-decoration: none;"
      >
        <div
          class="MuiCardContent-root"
        >
          <div
            class="stats"
          >
            <h6
              class="MuiTypography-root owner MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
              data-testid="owner-name"
            >
              facebook
            </h6>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-1 17.93c-3.95-.49-7-3.85-7-7.93 0-.62.08-1.21.21-1.79L9 15v1c0 1.1.9 2 2 2v1.93zm6.9-2.54c-.26-.81-1-1.39-1.9-1.39h-1v-3c0-.55-.45-1-1-1H8v-2h2c.55 0 1-.45 1-1V7h2c1.1 0 2-.9 2-2v-.41c2.93 1.19 5 4.06 5 7.41 0 2.08-.8 3.97-2.1 5.39z"
              />
            </svg>
          </div>
          <h2
            class="MuiTypography-root title MuiTypography-h6 MuiTypography-noWrap"
            data-testid="repo-name"
          >
            react
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap MuiTypography-gutterBottom"
          >
            Library for building UIs
          </p>
          <div
            class="stats"
          >
            <h6
              class="MuiTypography-root typography MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
              150K
            </h6>
            <h6
              class="MuiTypography-root typography MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9.4 16.6L4.8 12l4.6-4.6L8 6l-6 6 6 6 1.4-1.4zm5.2 0l4.6-4.6-4.6-4.6L16 6l6 6-6 6-1.4-1.4z"
                />
              </svg>
              JavaScript
            </h6>
          </div>
          <h6
            class="MuiTypography-root pos MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
          >
            Updated on Invalid Date at Invalid Date
          </h6>
        </div>
      </a>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
